### ISSUE TITLE
mpi.mod error during compilation

### ISSUE THREAD

**Issue by RatulSabui** [Warwick-Plasma/epoch#419]
I was trying to compile the most recent version of EPOCH.
'make' threw an error while mpi_version.f90 compilation was in progress. Apparently it did not find the 'mpi.mod' file and thus could not 'USE mpi'.

#######################################################################
>make MAKEFLAGS= COMPILER=gfortran -C ../SDF/FORTRAN
>make[2]: Entering directory `/work4/clf/ratul/epoch-4.18_normal/SDF/FORTRAN'
>mpif90 -I/usr/include -Iinclude -Iobj -Jinclude -o obj/MPI_CHECK src/mpi_version.f90
>rc/mpi_version.f90:3:7:

    3 |   USE mpi
      |       1
>Fatal Error: Cannot open module file ‘mpi.mod’ for reading at (1): No such file or directory
>compilation terminated.
>make[2]: *** [MPI_CHECK] Error 1
>make[2]: Leaving directory `/work4/clf/ratul/epoch-4.18_normal/SDF/FORTRAN'
>make[1]: *** [../SDF/FORTRAN/include/sdf.mod] Error 2
>make[1]: Leaving directory `/work4/clf/ratul/epoch-4.18_normal/epoch2d'
>make: *** [commit] Error 2

######################################################################

to correct this, I created a file mpif.f90 within epoch2d/src with  just the header file 'mpif.h' being included. following are the contents of the file that I created.

####################################################################
>MODULE MPI
include 'mpif.h'
END MODULE MPI

####################################################################

Also in the list of src files within epoch/epoch2d/Makefile, I added the 'mpif.f90' first so that this compilation happens before the rest of the files get compiled.

Now it threw up an error where some type declarations within mpif.h were deemed non-standard.

Am I missing something here? How do I diagnose this issue?

following are the versions in use:
epoch 4.18
GNU Fortran (GCC) 10.3.0
IBM Platform MPI Standard Edition: 09.01.04.03 [BASELINE_MPI_STREAM_REL_9.1.4.0_RTM_2015_06_15]  [] [] [] [] [] [] Linux x86-64

**Comment by Status-Mirror** [comment]
Hi Ratul,

Sometimes you get problems like this when you try to compile an updated version of EPOCH (weird .mod files).

Can you try running:
make clean

And then try to compile the code as normal? If you've been changing the source code, you may want to pull an unedited version first.

Cheers,
Stuart

**Comment by RatulSabui** [comment]
Thanks for your reply Stuart,

Did a clean. Did try recompiling an unedited version of the code. Same error comes up.

regards,
Ratul.

> Hi Ratul,
> 
> Sometimes you get problems like this when you try to compile an updated version of EPOCH (weird .mod files).
> 
> Can you try running: make clean
> 
> And then try to compile the code as normal? If you've been changing the source code, you may want to pull an unedited version first.
> 
> Cheers, Stuart

**Comment by Status-Mirror** [comment]
Hi Ratul,

Unfortunately I'm away from my workspace for a few weeks, so I can't do a proper debug.

Final quick thing to try - it looks like the error is pointing at an SDF file. To do a full clean which also resets SDF, you can try:
make cleanall

If this doesn't work, I'll be able to do more in a few weeks.

Cheers,
Stuart

**Comment by RatulSabui** [comment]
Update on this issue:

Somehow the code got compiled(with no special effort over my previous attempts). but then while executing the binary, it threw up the following error:

error: unknown option -pmi_args.

I guess the compilation is still not perfect.

**Comment by Status-Mirror** [comment]
I'm going to mark this issue closed for now, but feel free to continue the thread if you're still having problems.

**Comment by RatulSabui** [comment]
Sorry for the late response.

I did a module purge 
and then loaded the following modules for EPOCH4.18

intel 20.0.0 
intelmpi 20.0.0

The compilation was successful.

But at this moment I am facing problems using my older job-scripts with my new compilation.
I use a slurm based system where parallel jobs are initiated as:

mpirun -srun

For some reason, whenever I  use my newly compiled epoch Binary, the system is unable to recognize '-srun'.
I have tried explicily loading the modules in my job submission script but that hasnt been helpful.

I shall try more options and keep udating this thread.
